modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,375 @@
+
+	if ( ! empty( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	} else {
+		$metadata['has_transparency'] = false;
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_has_transparency_metadata', 10, 2 );
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @param array  $attr Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ *
+ * @return mixed
+ * @since n.e.x.t
+ */
+function dominant_color_tag_add_adjust_to_image_attributes( $attr, $attachment ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment->ID );
+	if ( ! is_array( $image_meta ) ) {
+		return $attr;
+	}
+
+	$has_transparency = isset( $image_meta['has_transparency'] ) ? $image_meta['has_transparency'] : false;
+
+	$extra_class = '';
+	if ( ! isset( $attr['style'] ) ) {
+		$attr['style'] = ' --has-transparency: ' . $has_transparency . '; ';
+	} else {
+		$attr['style'] .= ' --has-transparency: ' . $has_transparency . '; ';
+		$extra_class    = ' has-transparency ';
+	}
+
+	if ( isset( $image_meta['dominant_color'] ) ) {
+
+		$attr['data-dominant-color'] = $image_meta['dominant_color'];
+
+		if ( empty( $attr['style'] ) ) {
+			$attr['style'] = '';
+		}
+		$attr['style'] .= '--dominant-color: #' . $image_meta['dominant_color'] . ';';
+
+		$extra_class .= ( dominant_color_color_is_light( $image_meta['dominant_color'] ) ) ? 'dominant-color-light' : 'dominant-color-dark';
+		if ( empty( $attr['class'] ) ) {
+			$attr['class'] = '';
+		}
+		if ( isset( $attr['class'] ) && ! array_intersect( explode( ' ', $attr['class'] ), explode( ' ', $extra_class ) ) ) {
+			$attr['class'] .= ' ' . $attr['class'];
+		}
+	}
+
+	return $attr;
+}
+
+add_filter( 'wp_get_attachment_image_attributes', 'dominant_color_tag_add_adjust_to_image_attributes', 10, 2 );
+
+/**
+ * Filter image tags in content to add the dominant color to the image.
+ *
+ * @param string $filtered_image The filtered image.
+ * @param string $context The context of the image.
+ * @param int    $attachment_id The attachment ID.
+ *
+ * @return string image tag
+ * @since n.e.x.t
+ */
+function dominant_color_tag_add_adjust( $filtered_image, $context, $attachment_id ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment_id );
+
+	if ( ! is_array( $image_meta ) || ! isset( $image_meta['dominant_color'], $image_meta['has_transparency'] ) ) {
+
+		return $filtered_image;
+	}
+
+	/**
+	 * Filters whether dominant color is added to the image.
+	 * set to false inorder disable adding the dominant color to the image.
+	 *
+	 * @param bool   $add_dominant_color_to_image
+	 * @param int    $attachment_id
+	 * @param array  $image_meta
+	 * @param string $filtered_image
+	 * @param string $context
+	 */
+	if ( apply_filters( 'enable_dominant_color_for_image', true, $attachment_id, $image_meta, $filtered_image, $context ) ) {
```

Comment:
Let's align this better with the similar core filters, and also we need to align this more with the name of the function (see related comment above). It also needs to get the module prefix (`dominant_color`). So based on the above, this could be `dominant_color_img_tag_add_dominant_color`.

We should also align the order of the parameters for better consistency with the existing core filters for similar areas. For example like this:
```suggestion
	if ( apply_filters( 'dominant_color_img_tag_add_dominant_color', true, $filtered_image, $context, $attachment_id ) ) {
```

Image meta isn't passed in core and doesn't need to be, so let's also not do that here.